api_app/analyzers_manager/observable_analyzers/dns/dns_malicious_detectors/mullvad_dns.py

```diff
@@ -0,0 +1,128 @@
+    to check a domain's DNS records. It supports two modes:
+      - "query": returns raw DNS answer data.
+      - "malicious": interprets an NXDOMAIN (rcode==3) as the domain being blocked (i.e., malicious).
+    """
+
+    url = "https://base.dns.mullvad.net/dns-query"
+
+    def update(self):
+        pass
+
+    @staticmethod
+    def encode_query(observable: str) -> str:
+        """
+        Constructs a DNS query for the given observable (domain) for an A record,
+        converts it to wire format, and encodes it in URL-safe base64.
+        """
+        logger.info(f"Encoding DNS query for {observable}")
+        query = dns.message.make_query(observable, dns.rdatatype.A)
+        wire_query = query.to_wire()
+        encoded_query = (
+            base64.urlsafe_b64encode(wire_query).rstrip(b"=").decode("ascii")
+        )
+        logger.info(f"Encoded query: {encoded_query}")
+        return encoded_query
+
+    def run(self):
+        """
+        Executes the analyzer:
+          - Validates the observable type (DOMAIN or URL).
+          - For URLs, extracts the hostname.
+          - Encodes a DNS "A" record query.
+          - Makes an HTTP GET request to the Mullvad DoH endpoint.
+          - Parses the DNS response.
+          - Depending on the configured mode ("query" or "malicious"), returns either raw data or a flagged result.
+        """
+        observable = self.observable_name
+
+        if self.observable_classification == Classification.URL:
+            logger.info(f"Extracting hostname from URL {observable}")
+            hostname = urlparse(observable).hostname
+            observable = hostname
+
+        encoded_query = self.encode_query(observable)
+        complete_url = f"{self.url}?dns={encoded_query}"
+        logger.info(f"Requesting Mullvad DNS at: {complete_url}")
+
+        try:
+            response = httpx.Client(http2=True).get(
+                complete_url,
+                headers={"accept": "application/dns-message"},
+                timeout=30.0,
+            )
+            response.raise_for_status()
+        except httpx.HTTPError as e:
+            logger.error(f"HTTP error: {e}")
+            raise AnalyzerConfigurationException(f"Failed to query Mullvad DNS: {e}")
+
+        dns_response = dns.message.from_wire(response.content)
+
+        if self.mode == "malicious":
+            if dns_response.rcode() == 3:
+                return malicious_detector_response(
+                    observable=observable,
+                    malicious=True,
+                    note="Domain is blocked by Mullvad DNS content filtering.",
+                )
+            else:
+                return malicious_detector_response(
+                    observable=observable,
+                    malicious=False,
+                    note="Domain is not blocked by Mullvad DNS content filtering.",
+                )
+
+        else:

```

Comment:
Here there should be a check on the value like
```suggestion
        elif self.mode == "query":
```

Then you can add a `else` clause to handle all other invalid values.